File: Tatsuki_Yabe/Q3/Go/solution.go
package main

import "sort"

func numRescueBoats(people []int, limit int) int {
	sort.Ints(people)

	heavyP := len(people) - 1
	lightP := 0
	boatNum := 0

	for lightP <= heavyP {
		if people[lightP]+people[heavyP] <= limit {
			boatNum += 1
			lightP += 1
			heavyP -= 1
		} else {
			boatNum += 1
			heavyP -= 1
		}
	}

	return boatNum
}

File: Tatsuki_Yabe/Q3/Typescript/solution.ts
function numRescueBoats(people: number[], limit: number): number {
  people.sort();

  var heavyP = people.length - 1;
  var lightP = 0;
  var boatNum = 0;

  while (lightP <= heavyP) {
    if (people[lightP] + people[heavyP] <= limit) {
      boatNum++;
      lightP++;
      heavyP--;
    } else {
      boatNum++;
      heavyP--;
    }
  }

  return boatNum;
}
